### PR TITLE
Add Support back for Ubuntu as OS

### PR DESCRIPTION
**Pull request by mrueg**
**Description of the change**
Picking up https://github.com/onedr0p/flux-cluster-template/pull/527 and https://github.com/onedr0p/flux-cluster-template/issues/389

Adding support for Ubuntu
**Benefits**
More choice for users on the OS layer
**Possible drawbacks**
More maintenance (thus calling it out as  limited community support)

**Additional information**

I added the requested changes from https://github.com/onedr0p/flux-cluster-template/pull/527#pullrequestreview-1190466271

To do:

- [ ] Disable AppArmor (if that is a must-have)
- [ ] Test on a real machine

### CODE REVIEW

**Comment by onedr0p** [comment]
I would suggest to removing the options to disable swap and ufw and keep them disabled.

swap on Kubernetes isn't really supported and ufw could get in the way since kubernetes / CNIs wants to control iptables.

**Comment by mrueg** [comment]
I think it should be up to the user to enable / disable the features. With the right configuration, a host level firewall is still a good thing (ofc it should not have any conflicting rules). 
There's swap support in https://github.com/kubernetes/enhancements/issues/2400

**Comment by onedr0p** [comment]
> I think it should be up to the user to enable / disable the features.

That's fine but they can just remove the tasks they don't need or expand on them on their fork. This is afterall highly opinionated :)

I'd like to have those knobs removed because if a person doesn't understand the implications of enabling swap or ufw (and setting fw rules at the host level) they might end up having a bad time. Having those options force disabled without a knob to turn might get them to question why and how that is.

**Comment by mrueg** [comment]
I get that point, I would still prefer to have it, so I can reintegrate your upstream changes in my fork. Otherwise I also have to adjust patches etc., this approach keeps the delta minimal

File: provision/ansible/playbooks/cluster-prepare.yml [hunk 1]
```diff
@@ -85,14 +105,26 @@
...
+        - name: System Configuration (2) | Disable swap at runtime | Ubuntu
+          ansible.builtin.command: swapoff -a
+          when: ansible_facts['distribution'] == 'Ubuntu' and "{{ ubuntu.disable_swap | default true }}" == true and ansible_swaptotal_mb > 0
+        - name: System Configuration (2) | Disable swap at boot | Ubuntu
+          ansible.posix.mount:
+            name: "{{ item }}"
+            fstype: swap
+            state: absent
+            loop: ["none", "swap"]
+          when: ansible_facts['distribution'] == 'Ubuntu' and "{{ ubuntu.disable_swap | default true }}" == true
```
Comment (onedr0p) [inline provision/ansible/playbooks/cluster-prepare.yml:122]:
```suggestion
          when:
            - ansible_facts['distribution'] == 'Ubuntu'
            - "{{ ubuntu.disable_swap | default true }}" == true
```

File: provision/ansible/playbooks/cluster-prepare.yml [hunk 2]
```diff
@@ -85,14 +105,28 @@
...
-        - name: System Configuration (2) | Disable swap
+        - name: System Configuration (2) | Disable swap | Fedora
           ansible.builtin.dnf:
             name: zram-generator-defaults
             state: absent
-        - name: System Configuration (2) | Permissive SELinux
+          when: ansible_facts['distribution'] == 'Fedora'
+        - name: System Configuration (2) | Disable swap at runtime | Ubuntu
+          ansible.builtin.command: swapoff -a
+          when: ansible_facts['distribution'] == 'Ubuntu' and "{{ ubuntu.disable_swap | default true }}" == true and ansible_swaptotal_mb > 0
```
Comment (onedr0p) [inline provision/ansible/playbooks/cluster-prepare.yml:115]:
```suggestion
          when:
            - ansible_facts['distribution'] == 'Ubuntu'
            - "{{ ubuntu.disable_swap | default true }}" == true
            - ansible_swaptotal_mb > 0
```

File: provision/ansible/playbooks/cluster-prepare.yml [hunk 3]
```diff
@@ -44,12 +56,20 @@
...
             masked: true
             state: stopped
+          when: ansible_facts['distribution'] == 'Fedora'
+        - name: System Configuration (1) | Disable ufw | Ubuntu
+          ansible.builtin.systemd:
+            service: ufw.service
+            enabled: false
+            masked: true
+            state: stopped
+          when: ansible_facts['distribution'] == 'Ubuntu' and "{{ ubuntu.disable_ufw | default true }}" == true
```
Comment (onedr0p) [inline provision/ansible/playbooks/cluster-prepare.yml:72]:
```suggestion
          when:
            - ansible_facts['distribution'] == 'Ubuntu'
            - "{{ ubuntu.disable_ufw | default true }}" == true
```

**Review by onedr0p — APPROVED** [review]
Seems good for now, but I can't promise that the logic for ufw or swap will stay.

Thanks!

**Pull request merged**